operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/ResultCondition.java

```diff
@@ -0,0 +1,22 @@
+package io.javaoperatorsdk.operator.processing.dependent.workflow;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
+
+public interface ResultCondition<R, P extends HasMetadata, T> extends Condition<R, P> {
```

Comment:
Makes sense. Will do!

Reply:
Hmm, on second thought, it might make sense to leave it as-is because moving this to Condition would trigger needed changes to all existing conditions even if they don't have any interest in returning a result (unless, of course, we don't want to make it generic and just return Object as the result).

Reply:
This isn't comparable to `Matcher.Result`: this isn't a result class, this is a class that can return a result so it is quite different.

Reply:
I meant the `Result` interface not the `ResultCondition`.

Reply:
I still think that we should not complicate things and have a separate `ResultCondition` and `Condition`, since we work on v5, putting this directly to condition is completely fine IMO, and can be easily migrated in a controller simply one line code change.

Reply:
I'd do that in a second step if you feel it's important. I'll let you check the impact on existing code first (hint: it's not a one line change 😉).